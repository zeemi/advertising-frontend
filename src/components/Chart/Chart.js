import React from 'react';
import {CartesianGrid, Legend, Line, LineChart, Tooltip, XAxis, YAxis} from "recharts";
import './Chart.css';

const Chart = ({metrics, isFetching}) => {

  return <div className={'Chart'}>
    <LineChart width={800} height={500} data={metrics}
               margin={{top: 5, right: 30, left: 20, bottom: 5}}>
      <XAxis/>
      <YAxis width={100} yAxisId='left' label={{value: 'Clicks', angle: -90, dx: -20}}/>
      <YAxis width={100} yAxisId='right' orientation='right'
             label={{value: 'Impressions', angle: 90, margin: 100, dx: 60}}/>
      <CartesianGrid strokeDasharray="3 3"/>
      <Tooltip/>
      <Legend/>
      <Line yAxisId='left' type="monotone" stroke="#8884d8" dataKey={'clicks'} dot={false}/>
      <Line yAxisId='right' type="monotone" stroke="#82ca9d" dataKey={'impressions'} dot={false}/>
    </LineChart>
    {isFetching ? 'loading...' : null}

  </div>;
};

export default Chart;