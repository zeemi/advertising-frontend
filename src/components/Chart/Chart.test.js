import React from 'react';
import {shallow} from 'enzyme';
import Chart from './Chart';

describe('<Chart>', function () {
  it('should render chart using given data', function () {
    expect(shallow(<Chart metrics={[]}/>)).toMatchSnapshot()
  });
});