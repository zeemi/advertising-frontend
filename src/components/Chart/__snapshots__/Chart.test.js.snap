// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart> should render chart using given data 1`] = `
<div
  className="Chart"
>
  <LineChart
    barCategoryGap="10%"
    barGap={4}
    data={Array []}
    height={500}
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 20,
        "right": 30,
        "top": 5,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
    width={800}
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      label={
        Object {
          "angle": -90,
          "dx": -20,
          "value": "Clicks",
        }
      }
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="number"
      width={100}
      yAxisId="left"
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      label={
        Object {
          "angle": 90,
          "dx": 60,
          "margin": 100,
          "value": "Impressions",
        }
      }
      mirror={false}
      orientation="right"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      type="number"
      width={100}
      yAxisId="right"
    />
    <CartesianGrid
      fill="none"
      horizontal={true}
      horizontalFill={Array []}
      horizontalPoints={Array []}
      stroke="#ccc"
      strokeDasharray="3 3"
      vertical={true}
      verticalFill={Array []}
      verticalPoints={Array []}
    />
    <Tooltip
      active={false}
      allowEscapeViewBox={
        Object {
          "x": false,
          "y": false,
        }
      }
      animationDuration={400}
      animationEasing="ease"
      contentStyle={Object {}}
      coordinate={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      cursor={true}
      cursorStyle={Object {}}
      filterNull={true}
      isAnimationActive={true}
      itemStyle={Object {}}
      labelStyle={Object {}}
      offset={10}
      separator=" : "
      useTranslate3d={false}
      viewBox={
        Object {
          "x1": 0,
          "x2": 0,
          "y1": 0,
          "y2": 0,
        }
      }
      wrapperStyle={Object {}}
    />
    <Legend
      align="center"
      iconSize={14}
      layout="horizontal"
      verticalAlign="bottom"
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="clicks"
      dot={false}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="#8884d8"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId="left"
    />
    <Line
      activeDot={true}
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="impressions"
      dot={false}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="#82ca9d"
      strokeWidth={1}
      type="monotone"
      xAxisId={0}
      yAxisId="right"
    />
  </LineChart>
</div>
`;
